engine/table/src/main/java/io/deephaven/engine/table/impl/QueryTable.java

```diff
@@ -1068,7 +1068,11 @@ void doRefilter(
                 final WhereListener.ListenerFilterExecution filterExecution =
                         listener.makeFilterExecution(source.getRowSet().copy());
                 filterExecution.scheduleCompletion(
-                        fe -> completeRefilterUpdate(listener, upstream, update, fe.addedResult));
+                        () -> completeRefilterUpdate(listener, upstream, update, filterExecution.addedResult),
+                        exception -> {
+                            // ignore? Shouldn't we notify the listeners of the failure?
```

Comment:
As discussed in your overview, yes we need to propagate the error to table listeners. I would create a method analogous to the `completeRefilterUpdate` which notifies listeners of the error. The implementation must also `upstream.release()` (see `completeRefilterUpdate`).